src/System.Numerics.Vectors/src/System/Numerics/Matrix4x4.cs

```diff
@@ -896,11 +896,11 @@ public static Matrix4x4 CreatePerspectiveFieldOfView(float fieldOfView, float as
             result.M21 = result.M23 = result.M24 = 0.0f;
 
             result.M31 = result.M32 = 0.0f;
-            result.M33 = farPlaneDistance / (nearPlaneDistance - farPlaneDistance);
+            result.M33 = float.IsPositiveInfinity(farPlaneDistance) ? -1.0f : farPlaneDistance / (nearPlaneDistance - farPlaneDistance);
```

Comment:
Style nit: I'd prefer to assign this intermediate result ("float.IsPositiveInfinity(farPlaneDistance) ? -1.0f : farPlaneDistance / (nearPlaneDistance - farPlaneDistance);") to a local which is then copied to .M33.  The later write to .M43 can then reuse the local rather than having to read back one of the output structure fields.